python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2394,6 +2399,71 @@ def get_handlers(self) -> Sequence[logging.Handler]:
     def store_event(self, event: "EventLogEntry") -> None:
         self._event_storage.store_event(event)
 
+    def should_retry_run(self, run: DagsterRun) -> bool:
```

Comment:
I think maybe this name could be a little clearer about when and why it is called - I would think based on the name that it is something that you can call at any time on a run and it will tell for you whether it can be retried, but what it actually is (as I understand it) is a method that you call exactly once at the moment that the run fails. The result of that function is then put on a tag on the run, and after that, that tag is what you should use to determine whether or not the run is going to retry (and is what the actual retry logic in the daemon will use).

Reply:
renamed to `auto_reexecution_should_retry_run`, but that still doesn't get to your point about only calling this fn once to determine  the tag value